opencog/query/PatternTerm.h

```diff
@@ -54,14 +54,16 @@ namespace opencog {
 
 class PatternTerm;
 typedef std::shared_ptr<PatternTerm> PatternTermPtr;
+typedef std::weak_ptr<PatternTerm> PatternTermWPtr;
 typedef std::vector<PatternTermPtr> PatternTermSeq;
+typedef std::vector<PatternTermWPtr> PatternTermWSeq;
 
 class PatternTerm
 {
 	protected:
 		Handle _handle;
 		PatternTermPtr _parent;
-		PatternTermSeq _outgoing;
+		PatternTermWSeq _outgoing;
```

Comment:
I thought about it, but realized if parent is weak, there's no strong shared_ptr keeping the `root_term` in memory.

Maybe there are other ways to do it.